todo-app/src/Components/todoItem.js

```diff
@@ -0,0 +1,94 @@
+//jshint esnext: true
+
+import React, { Component, Fragment } from 'react'
+
+import './styles/todoItem.css';
+import Button from './Button';
+import InputField from './InputField';
+
+
+export default class TodoItem extends Component {
+
+    handleRemove = () => { 
+        this.props.handleRemove(this.props.data.id);
+    }
+
+    handleToggleEdit = () => { 
+        this.props.handleToggleEdit(this.props.data.id);
+    }
+
+    handleUpdate = () => {
+        this.props.handleUpdate(this.props.data.id);
+    }
+
+    handleChangeItem = event => {
+        this.props.handleChangeItem(event);
+    }
+
+    handleChangeDeadline = event => {
+        this.props.handleChangeDeadline(event);
+    }
+    
+    toggleEditMode = () => {
+        if (!this.props.isEditing) {
+            return (
+                <Button buttonContent='Edit' onClick={this.handleToggleEdit}/>
+            )
+        } else {
+            return (
+                <Fragment>  {/*Remember this is not understood why!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!*/}
```

Comment:
What do you mean here?

Reply:
Yeah this, since this is being rendered inside a div (in my `render`) Why do I still need to wrap it?? Is it not like only wrap it if it's in render > return???

Reply:
Ah, I see what you mean. No, the adjacent sibling issue is not specific to the `render` function.

There is nothing special about the `render` function that tells React "this is the place where JSX will be interpreted".

Your `toggleEditMode` function is not returning a string... Your JSX is not "copy/pasted" into place inside your `render` function, it must be interpreted as the function returns.

This question is a little bit outside of the box because returning JSX from class isn't a standard procedure. Of course this is all a matter of opinion, but I would say that if you need a secondary function to return JSX, you should consider creating a new component and put that logic inside the new component's `render` function. To me, the presence of a function like `toggleEditMode` returning JSX indicates that there is probably a component you can abstract out there. Again, it's all opinion.

*The short answer:* the adjacent sibling rule applies wherever JSX is present.

Another option would be to return your children here in an array, but then they need to be comma-separated and have keys:

    toggleEditMode = () => {
        if (!this.props.isEditing) {
            return (
                <Button buttonContent='Edit' onClick={this.handleToggleEdit}/>
            )
        } else {
            return [
              <Button key="toggleEditMode-updateButton" buttonContent='Update' onClick={this.handleUpdate} className='updateButton' />,
              <Button key="toggleEditMode-cancelButton" buttonContent='Cancel' onClick={this.props.handleCancelEdit} className='cancelButton' />
            ]
        }
    }